zebra-consensus/src/primitives/redjubjub.rs

```diff
@@ -10,11 +10,44 @@ use std::{
     task::{Context, Poll},
 };
 
+use futures::future::{ready, Ready};
+use once_cell::sync::Lazy;
 use rand::thread_rng;
 use redjubjub::{batch, *};
 use tokio::sync::broadcast::{channel, RecvError, Sender};
 use tower::Service;
-use tower_batch::BatchControl;
+use tower_batch::{Batch, BatchControl};
+use tower_fallback::Fallback;
+use tower_util::ServiceFn;
+
+/// Global batch verification context for RedJubjub signatures.
+///
+/// This service transparently batches contemporaneous signature verifications,
+/// handling batch failures by falling back to individual verification.
+///
+/// Note that making a `Service` call requires mutable access to the service, so
+/// you should call `.clone()` on the global handle to create a local, mutable
+/// handle.
```

Comment:
Can we make this static private inside of a pub function that does the cloning for users? That way people just call `get_service()` or w/e to get a copy and they can't fuck it up on accident.

Reply:
There's no way for them to screw it up because the static is immutable, so the only way to use it is by calling `.clone()`.  The note is just to avoid confusion and point them in the right direction.

Reply:
Aah, yea, for some reason I thought there was a lock. I think it might still be better to point them in the right direction by exposing an interface that encourages the needed pattern rather than explaining it with a comment but I'll leave it up to you.